runtime/src/main/scala/com/soundcloud/twinagle/Server.scala

```diff
@@ -21,7 +21,9 @@ private[twinagle] class Server(endpoints: Seq[ProtoRpc]) extends Service[Request
           case Some(service) =>
             service(request).handle {
               case e: TwinagleException => errorResponse(e)
-              case e                    => errorResponse(new TwinagleException(e))
+              case e: CancelledRequestException =>
+                errorResponse(new TwinagleException(ErrorCode.Canceled, "Request canceled by client"))
```

Comment:
nit: `TwinagleException` is a case class; you don't need the `new`
```suggestion
                errorResponse(TwinagleException(ErrorCode.Canceled, "Request canceled by client"))
```